_data-prepper/peer-forwarder.md

```diff
@@ -0,0 +1,181 @@
+| `drain_timeout`  | Duration | Represents the wait time for the Peer Forwarder to complete processing data before shutdown. |
+
+## SSL configuration
+
+The following SSL configuration table provides optional SSL configuration values and allows you to set up trust manager for the peer forwarding client to connect to other Data Prepper instances.
+
+| Value | Type | Description |
+| ----- | ---- | ----------- |
+| `ssl` | Boolean | Enables TLS/SSL. Default value is `true`. |
+| `ssl_certificate_file`| String | Represents the SSL certificate chain file path or Amazon Simple Storage Service (Amazon S3) path. The following is an example of an Amazon S3 path: `s3://<bucketName>/<path>`. Defaults to the default certificate file,`config/default_certificate.pem`. See [Default Certificates](https://github.com/opensearch-project/data-prepper/tree/main/examples/certificates) for more information about how the certificate is generated. |
+| `ssl_key_file`| String | Represents the SSL key file path or AWS S3 path. S3 path example `s3://<bucketName>/<path>`. Defaults to `config/default_private_key.pem` which is the default private key file. Read more about how the private key file is generated at the [Default Certificates](https://github.com/opensearch-project/data-prepper/tree/main/examples/certificates) page. |
+| `ssl_insecure_disable_verification` | Boolean | Disables the verification of the server's TLS certificate chain. Default value is `false`. |
+| `ssl_fingerprint_verification_only` | Boolean | Disables the verification of the server's TLS certificate chain and instead verifies only the certificate fingerprint. Default value is `false`. |
+| `use_acm_certificate_for_ssl` | Boolean | Enables TLS/SSL using the certificate and private key from AWS Certificate Manager (ACM). Default value is `false`. |
+| `acm_certificate_arn`| String | Represents the ACM certificate Amazon Resource Name (ARN). The ACM certificate takes precedence over S3 or the local file system certificate. Required if `use_acm_certificate_for_ssl` is set to `true`. |
+| `acm_private_key_password` | String | Represents the ACM private key password that will be used to decrypt the private key. If it's not provided, a random password will be generated. |
+| `acm_certificate_timeout_millis` | Integer | Represents the timeout in milliseconds required for ACM to get certificates. Default value is `120000`. |
+| `aws_region` | String | Represents the AWS Region that uses `ACM`, `S3` or `AWS Cloud Map`. Required if `use_acm_certificate_for_ssl` is set to `true` or `ssl_certificate_file`. Also required when the `ssl_key_file` is set to the `AWS S3` path, or if `discovery_mode` is set to `aws_cloud_map`. |
```

Comment:
```suggestion
| `aws_region` | String | Represents the AWS Region that uses `ACM`, `S3`, or `AWS Cloud Map`. Required if `use_acm_certificate_for_ssl` is set to `true` or `ssl_certificate_file`. Also required when the `ssl_key_file` is set to the `AWS S3` path or if `discovery_mode` is set to `aws_cloud_map`. |
```

Reply:
Is the path actually named `AWS S3`?

Reply:
@engechas Can you confirm?

Reply:
No, the path would be something of the format `s3://<bucketName>/<path>`. More info here: https://github.com/opensearch-project/data-prepper/blob/main/docs/peer_forwarder.md#ssl

Reply:
@carolxob Ensure the items in code format reflect the actual names of what we're referencing. I noticed this one because the service is named "Amazon S3" and not "AWS S3".

Reply:
@engechas It would be helpful if you could review the text and suggest what it should say in order to be technically accurate. If `AWS S3` is wrong here, what should the text say instead?

Reply:
@natebower @carolxob the path that @engechas is listing here is what. you would use in regards to using the AWS CLI or in a configuration file. The service itself is called "Amazon S3" as Nathan mentioned and if you want to get really specific, Amazon refers to an S3 path as "a path-style URL." Is the ask here then to clarify the wording in regards to service names and terminology and or are we asking to add an example path here? I kinda feel like the latter wouldn't be necessarily anyways in talking about regions as they have a different format such as `us-west-2`

Reply:
@ariamarble For my purposes, I'm just looking for the text to be technically accurate, so yes, just to clarify the wording.

Reply:
@natebower ok cool. so then yeah I would think that "Amazon S3 path" would be technically accurate.

Reply:
> is set to the `AWS S3` path

I think you could reword this to:

> is configured to use an Amazon S3 URL.

Reply:
Taking another look, we use the term "Amazon S3 path" above (on line 117) for the `ssl_key_file` value. So perhaps:

> is configured to use an Amazon S3 path.